js/providers/Model.js

```diff
@@ -461,7 +461,7 @@ define(
 				// a respository. If it is dirty, prompt the user to save and exit.
 				if (this.currentConceptSet()) {
 					if (this.currentConceptSetSource() == 'repository') {
-						if (this.currentConceptSetDirtyFlag && this.currentConceptSetDirtyFlag.isDirty() && !confirm("Concept set changes are not saved. Would you like to continue?")) {
+						if (this.currentConceptSetDirtyFlag() && this.currentConceptSetDirtyFlag().isDirty() && !confirm("Concept set changes are not saved. Would you like to continue?")) {
```

Comment:
Btw, this.currentConceptSetDirtyFlag() is guarunteed to be non-null (it's initalized at application startup) so, you shouldn't need to do the existance check, but  there's no harm in doing it. Just FYI.